pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -232,6 +238,8 @@ type Controller struct {
 	// CIDR ranger based on path-compressed prefix trie
 	ranger cidranger.Ranger
 
+	// Network name for to be used when the meshNetworks for registry nor network label on pod is specified
+	network string
```

Comment:
I would suggest using only one network

Reply:
Can you elaborate? Having them set in different places makes it clear which one takes precedence.

Reply:
I mean we should keep only one network field. It is duplicate with `networkForRegistry`.  As you said `networkForRegistry` takes precedence, always set it with the value in meshnetworks and if it is empty set it according to the ns label.

Reply:
That doesn't handle:
set meshNetworks
se label
unset meshNetworks